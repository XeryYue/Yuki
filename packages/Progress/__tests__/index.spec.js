import Progress from '../index'
import { mount } from '@vue/test-utils'

describe('Progress', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Progress)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be support different types', () => {
    const types = ['default', 'success', 'warning', 'error']
    const _default = mount(<Progress type={types[0]} value={10} />)
    const _success = mount(<Progress type={types[1]} value={10} />)
    const _warning = mount(<Progress type={types[2]} value={10} />)
    const _error = mount(<Progress type={types[3]} value={10} />)
    expect(_default.html()).toMatchSnapshot()
    expect(_success.html()).toMatchSnapshot()
    expect(_warning.html()).toMatchSnapshot()
    expect(_error.html()).toMatchSnapshot()
  })

  it('props colors should be work', () => {
    const colors = {
      20: 'red',
      40: 'blue',
    }
    const wrapper = mount(<Progress value={20} colors={colors} />)
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
})
