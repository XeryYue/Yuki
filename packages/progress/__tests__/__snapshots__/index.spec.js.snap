// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress props colors should be work 1`] = `
<div class="fect-progress">
  <div class="fect-progress__inner" title="20.00%" style="background-color: red; width: 20.00%;"></div><progress value="20" max="100"></progress>
</div>
`;

exports[`Progress should be support different types 1`] = `
<div class="fect-progress">
  <div class="fect-progress__inner" title="10.00%" style="width: 10.00%;"></div><progress value="10" max="100"></progress>
</div>
`;

exports[`Progress should be support different types 2`] = `
<div class="fect-progress">
  <div class="fect-progress__inner" title="10.00%" style="width: 10.00%;"></div><progress value="10" max="100"></progress>
</div>
`;

exports[`Progress should be support different types 3`] = `
<div class="fect-progress">
  <div class="fect-progress__inner" title="10.00%" style="width: 10.00%;"></div><progress value="10" max="100"></progress>
</div>
`;

exports[`Progress should be support different types 4`] = `
<div class="fect-progress">
  <div class="fect-progress__inner" title="10.00%" style="width: 10.00%;"></div><progress value="10" max="100"></progress>
</div>
`;
